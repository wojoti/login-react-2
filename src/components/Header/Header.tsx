import React, { ReactNode } from "react";
import { Wrapper } from "./Header.style";

type Props = {
  children?: ReactNode;
};

const Header = (props: Props) => {
  return <Wrapper>{props.children}</Wrapper>;
};

export default Header;
