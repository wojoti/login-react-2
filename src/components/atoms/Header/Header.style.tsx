import styled from "styled-components";
export const StyledHeader = styled.h1`
  font-size: 1rem;
  font-weight: 500;
  width: 100%;
`;
