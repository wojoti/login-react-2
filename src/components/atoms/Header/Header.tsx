import StyledHeader from "./Header.style";

interface Props {
  children?: React.ReactNode;
}

function Header({ children }: Props) {
  return <StyledHeader>{children}</StyledHeader>;
}

export default Header;
