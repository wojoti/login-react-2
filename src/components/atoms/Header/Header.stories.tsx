/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta, ComponentStory } from "@storybook/react";

import Header from "./Header";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Atoms/Header",
  component: Header,
} as ComponentMeta<typeof Header>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Header> = (args) => <Header {...args} />;

export const Login = Template.bind({});
Login.args = {
  children: "LOGIN",
};

export const SignUp = Template.bind({});
SignUp.args = {
  children: "SIGN UP",
};
