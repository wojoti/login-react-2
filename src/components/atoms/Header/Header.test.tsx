import { cleanup, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Header, { HeaderProps } from "./Header";

afterEach(() => {
  cleanup();
});

const props: HeaderProps = {
  testId: "test-header-id",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(
    <Header {...props}>test header</Header>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render header", () => {
  render(<Header {...props}>test header</Header>);
  const headerElement = screen.getByTestId("test-header-id");
  expect(headerElement).toBeInTheDocument();
});

test("should render header with text", () => {
  render(<Header {...props}>test header</Header>);
  const headerElement = screen.getByTestId("test-header-id");
  expect(headerElement).toBeInTheDocument();
  expect(headerElement).toHaveTextContent("test header");
});
