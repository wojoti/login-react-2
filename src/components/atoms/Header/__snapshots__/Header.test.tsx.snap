// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 1rem;
  font-weight: 500;
  width: 100%;
}

<h1
  className="c0"
  data-testid="test-header-id"
>
  test header
</h1>
`;
